core/asset/service.go

```diff
@@ -154,8 +148,10 @@ func (s *Service) DeleteAssets(ctx context.Context, request DeleteAssetsRequest)
 
 	if !request.DryRun && total > 0 {
 		newCtx, cancel := context.WithTimeout(context.Background(), s.config.DeleteAssetsTimeout)
-		s.cancelFnList = append(s.cancelFnList, cancel)
-		go s.executeDeleteAssets(newCtx, deleteSQLExpr)
+		go func() {
+			defer cancel()
```

Comment:
we don't really need to use `defer` here. just put the `cancel` after `s. executeDeleteAssets()` as they are just two lines